arangod/Cluster/ClusterCollectionMethods.cpp

```diff
@@ -878,11 +895,39 @@ ClusterCollectionMethods::createCollectionsOnCoordinator(
 
     // Now data contains the transaction;
     using namespace std::chrono_literals;
-    auto future = aac.sendWriteTransaction(120s, std::move(data))
-                      .thenValue(::reactToPreconditions)
-                      .thenValue(waitForOperationRoundtrip(ci));
-    return future.get();
+    auto res = aac.sendWriteTransaction(120s, std::move(data))
+                   .thenValue(::reactToPreconditions)
+                   .get();
 
+    if (res.fail()) {
+      return res.result();
+    }
+    // Now wait for the change to be happening
+    AgencyCallbackRegistry& callbackRegistry = ci.agencyCallbackRegistry();
+    auto currentGroup = pathCollectionGroupInCurrent(databaseName)
+                            ->group(std::to_string(col.groupID().id()))
+                            ->supervision()
+                            ->str(arangodb::cluster::paths::SkipComponents(1));
+    auto waitForSuccess = callbackRegistry.waitFor(
+        currentGroup, [versionToWaitFor](VPackSlice slice) {
+          if (slice.isNone()) {
+            // TODO: Should this actual set an "error"? It indicates
+            // that the collection is dropped if i am not mistaken
```

Comment:
If the collection group is dropped, is this going to hang? Or is there a timeout somewhere?
I imagine that (even if the slice is not none), the collection group could still be dropped while this is ongoing. We'll have to make sure there's no trouble caused because of that, otherwise this might be a tricky hanger to investigate. Not necessarily in this PR, but that could be perhaps a low-prio ticket.

Reply:
Absolutely. There are already other cases with exactly the same comment.
So this will be addressed in general.